lua/lazy/core/plugin.lua

```diff
@@ -110,7 +110,11 @@ function Spec:add(plugin, results)
     plugin.dev
     and (not Config.options.dev.fallback or vim.fn.isdirectory(Config.options.dev.path .. "/" .. plugin.name) == 1)
```

Comment:
will throw bug on this line if dev and fallback are true and dev.path is a function.

Instead, replace the dev plugin block with the following to achieve the desired effect

```lua
  -- dev plugins
  local function resolveDevPath(lazyPlugin)
    local path = nil
    if type(Config.options.dev.path == string)
      and vim.fn.isdirectory(Config.options.dev.path .. "/" .. lazyPlugin.name) == 1
    then
      path = Config.options.dev.path .. "/" .. lazyPlugin.name
    elseif type(Config.options.dev.path) == "function" then
      path = Util.norm(Config.options.dev.path(lazyPlugin))
      if type(path) ~= "string" then
        path = nil
      end
    end
    if Config.options.dev.fallback and path == nil then
      path = Config.options.root .. "/" .. lazyPlugin.name
    end
    return path
  end

  if
    plugin.dev
  then
    dir = resolveDevPath(plugin)
  elseif plugin.dev == false then
    -- explicitely select the default path
    dir = Config.options.root .. "/" .. plugin.name
  end
```